### PR TITLE
show different sensor types individually in report

### LINKED ISSUE #54: Report doesn't show bad channels
Bad channels are not annotated in the report

### PR DESCRIPTION
**Pull request by matsvanes**
Fixes #54 
@cgohil8 I tested this and it works well. I tested this for the case where you have either magnetometers, gradiometers or both, which result in one, two or three sensor topography plots respectively. 

The only thing I didn't know how to change - and I'm sure you know how to do so quickly - is make sure the size of the figures is good (i.e. when more than one topo is plotted, the figures in the report get small)

### CODE REVIEW

**Comment by matsvanes** [comment]
I don't know why this fails - it worked for me. It says the yml file with channel descriptions can't be found.
I put it in `/osl/utils/neuromag306_info.yml`
and refer to it like this:
```
        with open('utils/neuromag306_info.yml', 'r') as f:
            channels = yaml.load(f, Loader=yaml.FullLoader)
```

**Comment by cgohil8** [comment]
The figure size, i think the best way to fix this is just to create a smaller image with matplotlib, there's a figsize=(x,y) argument where x and y are sizes in inches.

For the file not found error, you have to specify the file relative to where the package is located. This is what I did for the parcellation files: https://github.com/OHBA-analysis/oslpy/blob/941d09d5ac592da3fab95049d781469ee6125a33/osl/source_recon/parcellation/parcellation.py#L41

File: osl/report/raw_report.py [hunk 1]
```diff
@@ -350,8 +350,21 @@ def plot_channel_time_series(raw, savebase=None):
 
 def plot_sensors(raw, savebase=None):
     """Plots sensors with bad channels highlighted."""
-
-    fig = raw.plot_sensors(show=False)
+    # plot channel types seperately for neuromag306 (3 coils in same location)
+    if 3012 in np.unique([i['coil_type'] for i in raw.info['chs']]):
+        with open('utils/neuromag306_info.yml', 'r') as f:
+            channels = yaml.load(f, Loader=yaml.FullLoader)
```
Comment (cgohil8) [inline osl/report/raw_report.py:356]: fyi, there's yaml.safe_load if you didn't want to specify the Loader

File: osl/utils/create_neuromag306_info.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                   'MEG1721', 'MEG1731', 'MEG1741', 'MEG1811', 'MEG1821', 'MEG1831', 'MEG1841', 'MEG1911', 'MEG1921',
+                   'MEG1931', 'MEG1941', 'MEG2011', 'MEG2021', 'MEG2031', 'MEG2041', 'MEG2111', 'MEG2121', 'MEG2131',
+                   'MEG2141', 'MEG2211', 'MEG2221', 'MEG2231', 'MEG2241', 'MEG2311', 'MEG2321', 'MEG2331', 'MEG2341',
+                   'MEG2411', 'MEG2421', 'MEG2431', 'MEG2441', 'MEG2511', 'MEG2521', 'MEG2531', 'MEG2541', 'MEG2611',
+                   'MEG2621', 'MEG2631', 'MEG2641']
+
+for i in channels.keys():
+    channels[i] = sorted(channels[i])
+
+with open('utils/neuromag306_info.yml', 'w') as outfile:
```
Comment (cgohil8) [inline osl/utils/create_neuromag306_info.py:98]: if the py file and yml file are in the same directory, then i think the following should be fine:
`str(Path(__file__).parent) + "neuromag306_info.yml"`

File: osl/report/raw_report.py (hunk 1 above)
Comment (matsvanes) [reply osl/report/raw_report.py]: Fixed in https://github.com/OHBA-analysis/oslpy/pull/55/commits/e9140485440b2828de8fdc0846c6dbfd6c4c389e

**Comment by matsvanes** [comment]
> The figure size, i think the best way to fix this is just to create a smaller image with matplotlib, there's a figsize=(x,y) argument where x and y are sizes in inches. Otherwise you need to modify the div element in the html to resize the figure.
> 
> For the file not found error, you have to specify the file relative to where the package is located. This is what I did for the parcellation files:
> 
> https://github.com/OHBA-analysis/oslpy/blob/941d09d5ac592da3fab95049d781469ee6125a33/osl/source_recon/parcellation/parcellation.py#L41

Fixed in https://github.com/OHBA-analysis/oslpy/pull/55/commits/e9140485440b2828de8fdc0846c6dbfd6c4c389e for the file path and https://github.com/OHBA-analysis/oslpy/pull/55/commits/dfdc1a212e13551dbd7d5875a9e12c4a4c106b63 for the figure size

**Pull request merged**